sdk/eventhub/azure-eventhub/azure/eventhub/_utils.py

```diff
@@ -262,3 +263,17 @@ def get_last_enqueued_event_properties(event_data):
         }
         return event_data._last_enqueued_event_properties
     return None
+
+def parse_sas_credential(credential):
+    # type: (AzureSasCredential) -> Tuple
+    try:
+        sas = credential.signature
+        parsed_sas = sas.split('&')
+        expiry = None
+        for item in parsed_sas:
+            if item.startswith('se='):
+                expiry = int(item[3:])
+        return (sas, expiry)
+    except AttributeError:
+        pass
+    return ()
```

Comment:
it what case would we encounter this AttributeError?

and if we `return ()`, I think `signature, expiry = parse_sas_credential(self._credential)` would report ValueError,
in this case shouldn't we raise a more reasonable error here like `ValueError("Invalid sas token")` ?

Reply:
technically if it's not an azuresascredential - we dont need the try except actually

Reply:
and the raise is inducing another client side validation - we must let the service handle this - updated

Reply:
agreed, sounds good to me